### PR TITLE
WTF: step 2 Add helpers

### PR DESCRIPTION
**Pull request by jasonLaster**
Associated Issue: #1511 

### Summary of Changes

* pulls the helpers into discrete files
* adds a mocha and mochi helper file. I think i can switch them in webpack which will be cool...

### Test Plan

mochitests pass

### CODE REVIEW

File: src/test/integration/tests/pretty-print.js [hunk 1]
```diff
@@ -29,9 +33,9 @@ async function prettyPrint () {
 
   invokeInTab("arithmetic");
   await waitForPaused(dbg);
-  assertPausedLocation(dbg, ppSrc, 18);
+  assertPausedLocation(dbg, ctx, ppSrc, 18);
```
Comment (jasonLaster) [inline src/test/integration/tests/pretty-print.js]: this doesn't feel great, but i have a feeling i can do better in the next phase

File: src/test/integration/utils/mochi.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+
+function info(msg) {
+  dump(`info: ${msg}\n`);
+}
+
+var ContentTask, gBrowser, openNewTabAndToolbox, Services, EXAMPLE_URL;
```
Comment (jasonLaster) [inline src/test/integration/utils/mochi.js]: I don't love this DI style, we hopefully can get this file priviledged like `head.js` with the proper permissions

File: src/test/mochitest/head.js [hunk 3]
```diff
@@ -41,6 +41,10 @@ Services.prefs.setBoolPref("devtools.debugger.new-debugger-frontend", true);
 Services.prefs.clearUserPref("devtools.debugger.tabs")
 Services.prefs.clearUserPref("devtools.debugger.pending-selected-location")
 
+this.gBrowser = gBrowser;
+this.Services = Services;
+this.EXAMPLE_URL = EXAMPLE_URL;
```
Comment (jasonLaster) [inline src/test/mochitest/head.js]: my fear here is that it's a memory leak. Also, ugly... but perhaps we can do something in another pass when we're cleaning up

File: src/test/integration/utils/wait.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+  await waitForState(dbg, state => {
+    const pause = dbg.selectors.getPause(state);
+    // Make sure we have the paused state.
+    if (!pause) {
+      return false;
+    }
+    // Make sure the source text is completely loaded for the
+    // source we are paused in.
+    const sourceId = pause.getIn(["frame", "location", "sourceId"]);
+    const sourceText = dbg.selectors.getSourceText(dbg.getState(), sourceId);
```
Comment (nchevobbe) [inline src/test/integration/utils/wait.js]: is `dbg.getState()` different from `state` ?

File: src/test/integration/utils/wait.js [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+      },
+      run: (dispatch, getState, action) => {
+        resolve(action);
+      }
+    });
+  });
+}
+
+// Wait until an action of `type` is dispatched. This is different
+// then `_afterDispatchDone` because it doesn't wait for async actions
```
Comment (nchevobbe) [inline src/test/integration/utils/wait.js]: nit: s/then/than

File: src/test/integration/utils/wait.js (hunk 4 above)
Comment (jasonLaster) [inline src/test/integration/utils/wait.js]: probably not... but i don't want to mess with it because we are only using this in one test now...

**Pull request merged**